### PR TITLE
LPS-190345: Fix NPE error when creating a sort with no existing EndPoint Id

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- For `API Sort` and `API Filter` we can create  them withou any assign `Endpoint`, so validate if there is any `API Application` related to them.

**JIRA Related Ticket**
https://liferay.atlassian.net/browse/LPS-190345

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
@sergiojimcos does this fulfill the expected result of the bug?

> Expected Results: The response return 400 and  message return 'An API Sortmust be related to a valid API endpoint.'

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
Validation needs to be performed in a model listener.

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.Objects;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Sergio Jiménez del Coso
+ */
+@Component(service = RelevantObjectEntryModelListener.class)
+public class APISortRelevantObjectEntryListener
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java:38]: @sergiojimcos this name does not match the interface name.

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+		if (objectEntry == null) {
+			return false;
+		}
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectEntry.getObjectDefinitionId());
+
+		if (!Objects.equals(
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java:74]: I think you can just `return !Objects.equals(objectDefinition.getExternalReferenceCode(), "L_API_ENDPOINT");`

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+	@Override
+	public void onBeforeUpdate(
+			ObjectEntry originalObjectEntry, ObjectEntry objectEntry)
+		throws ModelListenerException {
+
+		_validate(objectEntry);
+	}
+
+	private boolean _isValidAPIEndpoint(long apiSchemaId) throws Exception {
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java:62]: Rename `apiSchemaId` to `apiEndpointId`.

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	}
+
+	private void _validate(ObjectEntry objectEntry) {
+		try {
+			Map<String, Serializable> values = objectEntry.getValues();
+
+			long apiEndpointId = (long)values.get(
+				"r_apiEndpointToAPISorts_c_apiEndpointId");
+
+			if ((apiEndpointId != 0) && !_isValidAPIEndpoint(apiEndpointId)) {
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java:91]: What if `apiEndpointId === 0`?

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java (hunk 1 above)
Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryListener.java:38]: My fault, make sense!

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/138189

**Pull request closed without merging**